sapi/apache2handler/php_functions.stub.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+function apache_lookup_uri(string $filename): object|false {}
+
+function virtual(string $uri): bool {}
+
+function apache_request_headers(): array {}
+
+function getallheaders(): array {}
+
+function apache_response_headers(): array {}
+
+function apache_note(string $note_name, string $note_value = ""): string|false {}
```

Comment:
I'm not sure whether the default value of `$note_value` is indeed an empty string. If in doubt, use `UNKNOWN`.

Reply:
It looks nullable to me (https://github.com/php/php-src/blob/f826bbde93c55d08de9d962946d00dd0cf5d98a1/sapi/apache2handler/php_functions.c#L220) so I've updated the zpp call to allow null

Reply:
C `NULL` and PHP `null` are completely different things (the latter is a zval). To be actually nullable, the type specifier would have to be followed by `!`. I just checked again, and found that `apache_note` acts as getter if called with one argument, and as setter if called with two arguments. So either the default value should be `UNKNOWN`, or the ZPP format string should be adapted to also accept `null`. (Either way, the docs should be fixed as well.)

Reply:
Thanks for the help, I've reverted the changes and set it to `UNKNOWN` now, will submit a docs PR shortly

Reply:
Sorry, which part of the docs needs updating? It looks like it covers calling it with both one argument and two

Reply:
I was referring to the function signature, which is `apache_note ( string $note_name [, string $note_value = "" ] ) : string`. I'll fix that soonish.